### DISCUSSION TITLE
How to fix `Unsafe argument of type 'any' assigned to a parameter of type 'string'.` in Zod?

### DISCUSSION THREAD

**Discussion by lancejpollard** [colinhacks/zod#3150] [Q&A] [answered]
_Cross-posting b/c StackOverflow doesn't seem to have much of the zod community_

[How to fix `Unsafe argument of type 'any' assigned to a parameter of type 'string'.` in Zod?](https://stackoverflow.com/questions/77830312/how-to-fix-unsafe-argument-of-type-any-assigned-to-a-parameter-of-type-strin)

Thank you for the help.

**Comment by JacobWeisenburger** [comment, marked as answer]
try using `satisfies` like this:

```ts
const BuildCommandToDecompressWithUnarchiverModel = z.object( {
    overwrite: z.optional( z.boolean() ).default( false ),
    password: z.optional( z.string() ),
    output: z.object( {
        directory: z.object( {
            path: z.string(),
        } ),
    } ),
    input: z.object( {
        file: z.object( {
            path: z.string(),
        } ),
    } ),
} ) satisfies z.ZodType<BuildCommandToDecompressWithUnarchiver>
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
